### ISSUE TITLE
Simplejit gives a bus error

### ISSUE THREAD

**Issue by bjorn3** [CraneStation/cranelift#371]
<details>

```
warning: DefId(0/0:4 ~ example[8787]::callee[0]):

// WARNING: This output format is intended for human consumers only
// and is subject to change without notice. Knock yourself out.
fn callee() -> (){
    let mut _0: ();                      // return place

    bb0: {                              
        return;                          // bb0[0]: scope 0 at example.rs:46:15: 46:15
    }
}


function u0:8(i64) system_v {
    ss0 = explicit_slot 0

ebb0(v0: i64):
    jump ebb1

ebb1:
    return
}


warning: DefId(0/0:5 ~ example[8787]::call[0]):

// WARNING: This output format is intended for human consumers only
// and is subject to change without notice. Knock yourself out.
fn call() -> (){
    let mut _0: ();                      // return place

    bb0: {                              
        _0 = const callee() -> bb1;      // bb0[0]: scope 0 at example.rs:50:5: 50:13
                                         // ty::Const
                                         // + ty: fn() {callee}
                                         // + val: Value(Scalar(Bits { defined: 0, bits: 0 }))
                                         // mir::Constant
                                         // + span: example.rs:50:5: 50:11
                                         // + ty: fn() {callee}
                                         // + literal: const callee
    }

    bb1: {                              
        return;                          // bb1[0]: scope 0 at example.rs:51:2: 51:2
    }
}


function u0:10(i64) system_v {
    ss0 = explicit_slot 0
    sig0 = (i64) system_v
    fn0 = colocated u0:0 sig0

ebb0(v0: i64):
    jump ebb1

ebb1:
    call fn0(v0)
    jump ebb2

ebb2:
    return
}


./build.sh: line 3:  9418 Bus error: 10           rustc -Zcodegen-backend=$(pwd)/target/debug/librustc_codegen_cretonne.dylib example.rs --crate-type lib
```

</details>

This happens for all calls to a different function.

Edit: put output in `<details>`

**Comment by bjorn3** [comment]
Happens while calling `module.finish_all()`

**Comment by sunfishcode** [comment]
In callee's definition, callee is named u0:8, but in call, it's named u0:0. It's possible there's a bug in simplejit's symbol table, but it's also worth checking whether you're giving it the same name on both sides.

**Comment by bjorn3** [comment]
> In callee's definition, callee is named u0:8, but in call, it's named u0:0.

My fault. How do I get a `Function` with the correct `ExternalName` of a `FuncId`?

**Comment by bjorn3** [comment]
Using `Function::with_name_signature(ExternalName::user(0, func_id.index() as u32), sig)` still gives this error.

**Comment by bjorn3** [comment]
Looks like the issue is that finalize_function is not allowed to be called twice. https://github.com/bjorn3/rustc_codegen_cretonne/commit/dc323d7ffa9a5563f8c4f891992ac988db1f96fd fixes the issue. It would be nice to have some kind of a guard in simplejit against this though.

**Comment by sunfishcode** [comment]
I believe this is now fixed in #519. In fact, it's no longer possible to call `finalize_function` on an individual function. Please reopen or file new issues if there are any further concerns!
